launch/launch/event_handlers/on_process_exit.py

```diff
@@ -0,0 +1,105 @@
+# Copyright 2018 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Module for OnProcessExit class."""
+
+import collections
+from typing import Callable
+from typing import cast
+from typing import List
+from typing import Optional
+from typing import overload
+from typing import Text
+from typing import Tuple
+
+from ..event import Event
+from ..event_handler import EventHandler
+from ..events.process import ProcessExited
+from ..launch_context import LaunchContext
+from ..launch_description_entity import LaunchDescriptionEntity
+from ..some_actions_type import SomeActionsType
+
+
+if False:
+    # imports here would cause loops, but are only used as forward-references for type-checking
+    from ..actions import ExecuteProcess  # noqa
+
+
+class OnProcessExit(EventHandler):
+    """Convenience class for handling a process exited event."""
+
+    @overload
+    def __init__(
+        self, *,
+        target_action: 'ExecuteProcess' = None,
+        on_exit: SomeActionsType
+    ) -> None:
+        """Overload which takes just actions."""
+        ...
+
+    @overload  # noqa: F811
+    def __init__(
+        self,
+        *,
+        target_action: 'ExecuteProcess' = None,
+        on_exit: Callable[[int], Optional[SomeActionsType]]
+    ) -> None:
+        """Overload which takes a callable to handle the exit."""
+        ...
+
+    def __init__(self, *, target_action=None, on_exit) -> None:  # noqa: F811
+        """Constructor."""
+        from ..actions import ExecuteProcess  # noqa
```

Comment:
if you happen to remember why the noqa is needed here I'd be curious (it's used in the next line), but if not, don't worry about it

Reply:
Not sure, maybe a copy-paste that can be removed (it is required when being used with quoted type annotations only).

Reply:
The reason is that flake8 marks it as a redefinition of `ExecuteProcess` even though above it's imported under `if False`.